### PR TITLE
Remove public functions to match the stubs

### PR DESCRIPTION
**Pull request by jiegillet**
Related to [this analyzer check](https://github.com/exercism/elixir-analyzer/pull/207).

I have compared all the exemploids to the stubs and listed all the functions that were public in the exemploid but not in the stubs. Then I basically added a whole bunch of `p` until the problem went away. Some notes:

- Concept exercises: 3 functions have empty stubs, nothing we can do there except change the analyzer comment
- `circular-buffer`, `bank-account` and `react` have some GenServer `@impl true` functions that are not in the stubs but cannot be private. This is fine because students are expected to write those and their solution will be compared to the examples that have them too.
- `clock` implements its own `to_string` that is not in the stub, but required by the tests. Once again, this is fine for the same reason than above.
- `parallel-letter-frequency` is strange, I think it used to use `:rpc.pmap` but then it was commented and I don't think it's doing anything in parallel anymore. [My solution does](https://exercism.org/tracks/elixir/exercises/parallel-letter-frequency/solutions/jiegillet) if we want to replace it.
- I couldn't help but change some implementations a tiny little bit, sometimes using the notation `do_thing` because otherwise `thing` ends up with several arities. I also had to remove some unused functions.

### CODE REVIEW

File: exercises/practice/pythagorean-triplet/.meta/example.ex [hunk 1]
```diff
@@ -49,11 +49,9 @@ defmodule Triplet do
     multiples ++ generate(next ++ triplets, sum)
   end
 
-  @doc """
-  Given a primitive Pythagorean triple, generate three more.
-  [All primitive Pythagorean triples can be generated this way without duplication][https://en.wikipedia.org/wiki/Tree_of_primitive_Pythagorean_triples].
-  """
-  def next_triplets([a, b, c]),
```
Comment (jiegillet) [inline exercises/practice/pythagorean-triplet/.meta/example.ex:56]: Private functions cannot have docs, I just learned :)

**Review by jiegillet — COMMENTED** [review]
Here are some extra comments.

File: exercises/practice/parallel-letter-frequency/.meta/example.ex [hunk 2]
```diff
@@ -7,7 +7,7 @@ defmodule Frequency do
   def frequency(texts, workers) do
     groups = Enum.map(0..(workers - 1), &stripe(&1, texts, workers))
     # :rpc.pmap({Frequency, :count_texts}, [], groups)
-    Enum.map(groups, &Frequency.count_texts/1)
```
[the remark this answers is not in the document]
Comment (angelikatyborska) [reply exercises/practice/parallel-letter-frequency/.meta/example.ex:10]: Yup, this solution is not "parallel" in any way. Feel free to replace it with your own in a separate PR.

File: exercises/practice/pov/.meta/example.ex [hunk 3]
```diff
@@ -46,81 +46,53 @@ defmodule Pov do
...
-      }) do
-    %Zipper{
-      focus: left,
-      genealogy: [
-        %Crumb{crumb | left_siblings: lefties, right_siblings: [tree | right]} | grandparents
-      ]
-    }
-  end
-
-  def left(_zipper), do: nil
```
Comment (angelikatyborska) [inline exercises/practice/pov/.meta/example.ex:116]: Help me out here - did we have dead code in the example?

File: exercises/practice/sgf-parsing/.meta/example.ex [hunk 4]
```diff
@@ -175,19 +168,15 @@ defmodule SgfParsing do
...
-  def drop_and(p1, p2) do
+  defp drop_and(p1, p2) do
     fn string ->
       with {:ok, _, rest} <- p1.(string) do
         p2.(rest)
       end
     end
   end
 
-  def inject(a) do
```
Comment (angelikatyborska) [inline exercises/practice/sgf-parsing/.meta/example.ex:186]: `string`, `inject` - also dead code?

**Review by angelikatyborska — APPROVED** [review]

File: exercises/practice/pov/.meta/example.ex (hunk 3 above)
Comment (jiegillet) [reply exercises/practice/pov/.meta/example.ex:116]: Yes we did, that's my fault. Basically for this one and `sgf-parsing` I built a library (in this one, moving thought a tree with zipper, in the other a parser combinator) and the dead code is code that would definitely belong in the the library that I thought I would use but in the end I didn't.

File: exercises/practice/sgf-parsing/.meta/example.ex (hunk 4 above)
Comment (jiegillet) [reply exercises/practice/sgf-parsing/.meta/example.ex:186]: Yup. I was very surprised to realize I didn't need them in the end, but I didn't have the heart to delete them. Bad Jie!

Comment (angelikatyborska) [reply exercises/practice/sgf-parsing/.meta/example.ex:186]: Sounds like somebody didn't pay too much attention to the example solutions when doing the code review for the new exercises 😁

**Pull request merged**